src/platform/ESP32/ConfigurationManagerImpl.cpp

```diff
@@ -31,9 +31,9 @@
 #include <platform/ESP32/ESP32Config.h>
 #include <platform/internal/GenericConfigurationManagerImpl.ipp>
 
-#if CHIP_DEVICE_CONFIG_ENABLE_ETHERNET
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_ETHERNET) && CHIP_DEVICE_CONFIG_ENABLE_ETHERNET
```

Comment:
This is wrong.  CHIP_DEVICE_CONFIG_ENABLE_ETHERNET should always be defined in the device config.  If that's not happening, either someone messed up, or (worse) this file is not including the right config values.

I see that src/platform/ESP32/CHIPDevicePlatformConfig.h defines CHIP_DEVICE_CONFIG_ENABLE_ETHERNET only sometimes; should it be defining it to 0 in other cases, if this is an esp32-specific define?  If this is a general define, cross-platform, then CHIPDeviceConfig.h should be defaulting it, I assume.

Reply:
Fixed this